planner/core/optimizer.go

```diff
@@ -541,6 +542,23 @@ func prunePhysicalColumnsInternal(sctx sessionctx.Context, plan PhysicalPlan) er
 	return nil
 }
 
+// tryEnableLateMaterialization tries to push down some filter conditions to the table scan operator
+// @brief: push down some filter conditions to the table scan operator
+// @param: sctx: session context
+// @param: plan: the physical plan to be pruned
+// @note: this optimization is only applied when the TiFlash is used.
+// @note: the following conditions should be satisfied:
+//   - Only the filter conditions with high selectivity should be pushed down.
+//   - The filter conditions which contain heavy cost functions should not be pushed down.
+//   - Filter conditions that apply to the same column are either pushed down or not pushed down at all.
+func tryEnableLateMaterialization(sctx sessionctx.Context, plan PhysicalPlan) {
+	// check if EnableLateMaterialization is set
+	if sctx.GetSessionVars().EnableLateMaterialization {
+		logutil.BgLogger().Info("EnableLateMaterialization is set, try to push down some predicates to table scan.")
```

Comment:
The level here is too high, and there's no other information such as the table name. Or you can add it in the explain result as a warning output.